runtime/lua/vim/filetype.lua

```diff
@@ -174,9 +174,15 @@ end
 -- luacheck: push no unused args
 -- luacheck: push ignore 122
 
--- Filetypes based on file extension
+-- Filetype detection logic is encoded in three tables:
+-- 1. `extension` for literal extension lookup
+-- 2. `filename` for literal full filename or path lookup
```

Comment:
That's not quite right. `filename` is used with both [full path](https://github.com/neovim/neovim/blob/eb629cce917155a4d436327db10c8b1b44ae7861/runtime/lua/vim/filetype.lua#L2689) and [basename](https://github.com/neovim/neovim/blob/eb629cce917155a4d436327db10c8b1b44ae7861/runtime/lua/vim/filetype.lua#L2696).

Reply:
That's what "full filename" (as opposed to extension) "or path" refers to.